test/units/TEST-54-CREDS.sh

```diff
@@ -23,6 +23,45 @@ run_with_cred_compare() (
     diff "$log_file" <(echo -ne "$exp")
 )
 
+test_mount_with_credentials() {
+    local tmpdir credfile unit
+    tmpdir="/tmp/test-54-mount"
+    credfile="/tmp/mount-cred"
+    unit=$(systemd-escape --suffix mount --path "$tmpdir")
+
+    mkdir -p "$tmpdir"
+    echo foo >"$credfile"
+
+    # Set up test mount unit
+    cat >/run/systemd/system/"$unit" <<EOF
+[Mount]
+What=tmpfs
+Where=$tmpdir
+Type=tmpfs
+LoadCredential=loadcred:$credfile
+SetCredential=setcred:bar
+EOF
```

Comment:
this is not testing the creds at all.

any chance you can extend this? maybe invent your own binary /usr/sbin/mount.thisisatest, then mount a file system with Type=thisisatest, and inside of that script verify the the expected env var is set and the cred can be found.